### PR TITLE
clojure-lsp: 2023.08.06-00.28.06 -> 2024.04.22-11.50.26

### LINKED ISSUE #308790: Build failure: clojure-lsp
Hi folks,

I saw that clojure-lsp was out of date I tried to build the new version (2024.04.22-11.50.26)

Unfortunately after bumping the version and fixing the hashes, we have a build error. Specifically the version check that was recently added is failing.

### Steps To Reproduce

1. build clojure-lsp with version `2024.04.22-11.50.26` (see https://github.com/Ramblurr/nixpkgs/commit/832b07ac1c7d6bdef5598442e13e6d753b1ef160)

### Build log

Full build log: http://0x0.st/XXWG.log

```
...snip...
Finished generating 'clojure-lsp' in 19.8s.
@nix { "action": "setPhase", "phase": "checkPhase" }
Running phase: checkPhase
Exception in thread "main" java.lang.ExceptionInInitializerError
	at clojure.lang.Namespace.<init>(Namespace.java:34)
	at clojure.lang.Namespace.findOrCreate(Namespace.java:176)
	at clojure.lang.Var.internPrivate(Var.java:156)
	at clojure_lsp.main.<clinit>(Unknown Source)
	at java.base@22/java.lang.invoke.DirectMethodHandle.ensureInitialized(DirectMethodHandle.java:285)
	at java.base@22/java.lang.invoke.DirectMethodHandle.internalMemberNameEnsureInit(DirectMethodHandle.java:335)
Caused by: java.io.FileNotFoundException: Could not locate clojure/core__init.class, clojure/core.clj or clojure/core.cljc on classpath.
	at clojure.lang.RT.load(RT.java:462)
	at clojure.lang.RT.load(RT.java:424)
	at clojure.lang.RT.<clinit>(RT.java:338)
	... 6 more
```

### Notify maintainers

@ericdallo @babariviere @jcf 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.29, NixOS, 24.05 (Uakari), 24.05.20240502.63c3a29`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/p69bcs7ma6ijj8v9xsrg3nq3nn8ryn95-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Ramblurr**
fixes #308790 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 310027` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>clojure-lsp</li>
  </ul>
</details>

**Review by getchoo — APPROVED** [review]

File: pkgs/development/tools/misc/clojure-lsp/default.nix [hunk 1]
```diff
@@ -12,26 +12,27 @@
...
   };
 
   extraNativeImageBuildArgs = [
     "--no-fallback"
     "--native-image-info"
+    "--features=clj_easy.graal_build_time.InitClojureClasses"
   ];
 
-  doCheck = true;
+  doCheck = false;
```
Comment (pbsds) [inline pkgs/development/tools/misc/clojure-lsp/default.nix:35]: Is this still needed if you bump further? Can you s

**Review by ericdallo — APPROVED** [review]

File: pkgs/development/tools/misc/clojure-lsp/default.nix (hunk 1 above)
Comment (Ramblurr) [reply pkgs/development/tools/misc/clojure-lsp/default.nix:35]: What do you mean by "bump further"?

**Comment by Ramblurr** [comment]
Adding `--features=clj_easy.graal_build_time.InitClojureClasses` fixed the tests, which was a nice find.

File: pkgs/development/tools/misc/clojure-lsp/default.nix (hunk 1 above)
Comment (Ramblurr) [reply pkgs/development/tools/misc/clojure-lsp/default.nix]: Ah.. this is resolved anyways by `--features=clj_easy.graal_build_time.InitClojureClasses`

File: pkgs/development/tools/misc/clojure-lsp/default.nix [hunk 2]
```diff
@@ -12,23 +12,24 @@
...
   pname = "clojure-lsp";
-  version = "2023.08.06-00.28.06";
+  version = "2024.04.22-11.50.26";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
     rev = version;
-    sha256 = "sha256-wc7M2cPRtdaRzZn3GNu/aCbQ2VqxiDxvu/b7qnBVUBo=";
+    sha256 = "sha256-GyPIFYR+/BZ+vq6+yuer5HoVILXLWNw1sW8XpJ7q4SA=";
```
Comment (superherointj) [inline pkgs/development/tools/misc/clojure-lsp/default.nix:21]:
```suggestion
    hash = "sha256-GyPIFYR+/BZ+vq6+yuer5HoVILXLWNw1sW8XpJ7q4SA=";
```

File: pkgs/development/tools/misc/clojure-lsp/default.nix [hunk 3]
```diff
@@ -12,23 +12,24 @@
 
 buildGraalvmNativeImage rec {
   pname = "clojure-lsp";
-  version = "2023.08.06-00.28.06";
+  version = "2024.04.22-11.50.26";
 
   src = fetchFromGitHub {
     owner = pname;
     repo = pname;
```
Comment (superherointj) [inline pkgs/development/tools/misc/clojure-lsp/default.nix:19]:
```suggestion
    owner = "clojure-lsp";
    repo = "clojure-lsp";
```

**Comment by superherointj** [comment]
> meta.changelog = "https://github.com/clojure-lsp/clojure-lsp/releases/tag/${version}";

(This is used by the bot in update script)

**Comment by superherointj** [comment]
Consider moving package to `by-name`.

File: pkgs/development/tools/misc/clojure-lsp/default.nix [hunk 4]
```diff
@@ -12,23 +12,24 @@
...
     rev = version;
-    sha256 = "sha256-wc7M2cPRtdaRzZn3GNu/aCbQ2VqxiDxvu/b7qnBVUBo=";
+    sha256 = "sha256-GyPIFYR+/BZ+vq6+yuer5HoVILXLWNw1sW8XpJ7q4SA=";
   };
 
   jar = fetchurl {
     url = "https://github.com/clojure-lsp/clojure-lsp/releases/download/${version}/clojure-lsp-standalone.jar";
-    sha256 = "c301821ac6914999a44f5c1cd371d46b248fe9a2e31d43a666d0bc2656cfdd78";
+    sha256 = "sha256-dB16225A7L3nWplvqlal+5gho+LmqqVGPN9dfasKaPk=";
```
Comment (superherointj) [inline pkgs/development/tools/misc/clojure-lsp/default.nix:26]:
```suggestion
    hash = "sha256-dB16225A7L3nWplvqlal+5gho+LmqqVGPN9dfasKaPk=";
```

**Comment by Ramblurr** [comment]
@jlesquembre @superherointj thanks for the feedback.

In the first commit I applied:


    * change sha256 -> hash
    * removed withs
    * documented where the build args come from
    * removed pname usage
    * added meta.changelog

In the 2nd commit I moved the package to `pkgs/by-name` and applied `nixfmt`

**Comment by jlesquembre** [comment]
Result of `nixpkgs-review pr 310027` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clojure-lsp</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
  </ul>
</details>

**Review by jlesquembre — APPROVED** [review]

**Review by ericdallo — APPROVED** [review]

**Review by ericdallo — APPROVED** [review]

**Review by superherointj — APPROVED** [review]

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 310027` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>clojure-lsp</li>
    <li>vscode-extensions.betterthantomorrow.calva</li>
  </ul>
</details>

**Comment by superherointj** [comment]
I'm awaiting for the `x86_64-darwin` build.

**Review by JohnRTitor — APPROVED** [review]
LGTM

**Pull request merged**